marge/single_merge_job.py

```diff
@@ -64,8 +64,9 @@ def update_merge_request_and_accept(self, approvals):
 
             if _updated_sha == actual_sha and self._options.guarantee_final_pipeline:
```

Comment:
i'm not sure that this condition is right or not.
maybe it could be `if _updated_sha != actual_sha && self._options.guarantee_final_pipeline`

in my test with this patch, marge bot is still unwaited the final testing result.

Reply:
Thank you for your interest in the PR, @d3m3vilurr.

> in my test with this patch, marge bot is still unwaited the final testing result.

Do you have the "Pipelines must succeed" option checked in Gitlab project settings? In my understanding, the main point of the `--guarantee-final-pipeline` switch is to enforce the mentioned Gitlab merge strategy in the case when no actual changes happened to the MR commits, so the pipeline is triggered for the last time, as it must succeed.

Reply:
you are right. we didn't mark the `pipeline must succeed` in gitlab project setting.
because we want to push & merge without margebot in emergency time.

in my case, a bot started with `guarantee_final_pipeline`, when reassigned to a bot, a branch of MR was rebased with a target branch by a bot.
(a target branch has another commits, so technically, MR was changed)
but if a project doesn't have  `pipeline must succeed` option, the last checking CI would be always failed cuz bot didn't wait for this CI before merging.

So, I want to wait for CI result like another bot (such as bors-ng)